### ISSUE TITLE
Implement Google Batch Normalization to beat human on ImageNet classification task

### ISSUE THREAD

**Issue by futurely** [BVLC/caffe#1990]
Sergey Ioffe, Christian Szegedy. [Batch Normalization: Accelerating Deep Network Training by Reducing Internal Covariate Shift](http://arxiv.org/abs/1502.03167). arXiv:1502.03167 [cs.LG]

**Comment by ducha-aiki** [comment]
#1965

**Comment by williford** [comment]
The following PRs were merged: #3229 (for BatchNorm layer) and #3591 (for Scale and Bias layers).
